File: Aug14/Carl/Q7.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        i_new = 0
        for i_old in range(0, len(nums)):
            if(nums[i_old]!=val):
                nums[i_new] = nums[i_old]
                i_new += 1
        return i_new


s = Solution()
arr = [1,2,3,4,5,5,6]
print(s.removeElement(arr, 5))
print(arr)

File: 2018/Aug14/Troy/Q7.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int i_new = 0;
        int i_old = 0;
        int len = nums.length;
        for (; i_old < len; i_old++) {
            if (nums[i_old] != val) {
                nums[i_new] = nums[i_old];
                i_new++;
            }
        }
        return i_new;
    }
}
